### PR TITLE
Display warning on preview environment to notify users if page is Mozart

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves JIRA N/A

Overall changes
======
Displays a warning on the preview environment if a page is rendered by Mozart (meaning that it cannot pick up changes from branch/PR)

Code changes
======
Add an indicator component to the header/footer
Add `requestServiceChain` to the request context 

Screenshot 2024-10-11 at 12 31 00
Screenshot 2024-10-11 at 12 31 10

Testing
======
Locally - cannot test on preview for pages not routed directy to Simorgh as they are rendering the test.bbc.com version
Unit tests
Preview environment: indicator not displayed for pages routed directly to Simorgh

Helpful Links
======
_Add Links to useful resources related to this PR if applicable._

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

File: src/app/legacy/psammead/psammead-brand/src/index.jsx [hunk 1]
```diff
@@ -165,7 +166,30 @@ const StyledBrand = ({ linkId, product, serviceLocalisedName = null, svg }) => (
...
+  const [isPreview, setIsPreview] = useState(false);
+
+  useEffect(() => {
+    const hostnameMatchesPreview = window.location.hostname.match(/preview/g);
+    const isPreviewEnvironment =
+      hostnameMatchesPreview && hostnameMatchesPreview.length > 0;
+
+    setIsPreview(isPreviewEnvironment);
+  }, []);
+
```
Comment (HarveyPeachey) [inline src/app/legacy/psammead/psammead-brand/src/index.jsx]: Just need to double check that no CLS or LCP issues happen or any weird behaviour with child components, as using state might cause a re-render. 

Not actually sure if react does a re-render when you set an already false state value to false, hopefully it's clever enough, or you could be doubly safe and just set the state when isPreviewEnvironment is true

**Review by HarveyPeachey — APPROVED** [review]
LGTM

File: src/app/legacy/psammead/psammead-brand/src/index.jsx (hunk 1 above)
Comment (karinathomasbbc) [reply src/app/legacy/psammead/psammead-brand/src/index.jsx]: Gosh I never thought about LCP/CLS 😬 Luckily this is only on the preview environment, but all the same, it's best to check. I'll ensure not to set state unless preview env is true & see if that causes a re-render or not.

Comment (karinathomasbbc) [reply src/app/legacy/psammead/psammead-brand/src/index.jsx]: But of course this code will run regardless of which environment it's on. I'll do a quick comparison between the test environment & preview with this banner to see if there's a difference in Web Vital scores.

Comment (HarveyPeachey) [reply src/app/legacy/psammead/psammead-brand/src/index.jsx]: It should be fine, as I don't think there's anything that will actually make react change anything in the DOM, but best to double check

**Review by amoore108 — APPROVED** [review]

**Comment by karinathomasbbc** [comment]
Closing, as we're in the process of migrating all page types to Simorgh instead of via Mozart, so this will no longer be required.

**Pull request closed without merging**